llvm/lib/MCCAS/MCCASObjectV1.cpp

```diff
@@ -505,12 +505,13 @@ static Error materializeDebugInfoOpt(MCCASReader &Reader,
                           StringRef FormData, bool) {
     if (Form == dwarf::Form::DW_FORM_ref4_cas ||
         Form == dwarf::Form::DW_FORM_strp_cas) {
-      auto Reader = BinaryStreamReader(FormData, endianness::little);
-      uint64_t Data64;
-      if (auto Err = Reader.readULEB128(Data64))
-        handleAllErrors(std::move(Err));
+      DataExtractor Extractor(FormData, true, 8);
```

Comment:
Do we know for sure that there will be 8 bytes available at all times?

Reply:
That is just the address size, not how many bytes are available in the buffer